src/vs/workbench/browser/web.simpleservices.ts

```diff
@@ -950,7 +951,23 @@ export class SimpleWindowService implements IWindowService {
 		return Promise.resolve(undefined);
 	}
 
-	toggleFullScreen(): Promise<void> {
+	toggleFullScreen(target?: HTMLElement): Promise<void> {
+		if (!target) {
+			return Promise.resolve();
+		}
+
+		// Chromium, Safari and Edge 14 are all using webkit prefix.
+		if ((<any>document).webkitIsFullScreen !== undefined) {
+			if (!(<any>document).webkitIsFullScreen) {
+				(<any>target).webkitRequestFullscreen();
```

Comment:
@rebornix can this fail? Is there a way to find out that fullscreen worked?

Reply:
It can fail. For chromium/Firefox, it returns a valid promise so we can catch the error and turn into full screen mode only when it succeeds. For Safari, we don't know if it fails or not as it's an async task, but the api doesn't return a promise somehow ;(.